### ISSUE TITLE
Rotate mobile video 

### ISSUE THREAD

**Issue by palich12** [TadasBaltrusaitis/OpenFace#294]
Hello, Tadas!
 I find odd rotate in mobile video than have height > whith, so autorotate do imposible to recogonize face.
For examle:
input video http://chilp.it/9870f85
output video http://chilp.it/03618fd

How I can disable autorotate vide in OpenFace?

**Comment by TadasBaltrusaitis** [comment]
Interesting, OpenFace should not be doing any such rotation. This looks like a bug, I will explore this further.

What command line arguments did you use that produced such a result?

Thanks,
Tadas

**Comment by palich12** [comment]
Link to download in previos message
>FeatureExtraction.exe -vis-track -f input_mobile_video.mp4 -ov output.mp4 -of output.csv

**Comment by jgoenetxea** [comment]
The problem is not related with the code, but with the video itself. Some devices (like iPhones) store the frames of the video rotated, and then they define this rotation in the metadata. 
Many video players like VLC take into account the metadata, and they tend to rotate the image to show it properly to the user, but the [VideoCapture module of OpenCV does not rotate the image for you](https://stackoverflow.com/questions/44380432/opencv-video-looks-good-but-frames-are-rotated-90-degrees).
In this case, the video is stored rotated, and you need to apply a rotation if you want to process it.

**Comment by palich12** [comment]
Ok, thanks, I fix it by preprocessing video by command
>ffmpeg.exe -i input.mp4 -metadata:s:v:0 rotate=0 -y output.mp4

**Comment by TadasBaltrusaitis** [comment]
Thanks for the clarification

**Comment by s160421093** [comment]
can anyone help me how to use ffmpeg. The original video when opened using various applications such as media player, MPC-HC, windows media player and others looks portrait but when opened OpenFace becomes landscape. 

This video has a rotation metadata of -90 and also a width of 1920 and a height of 1080. I have tried replacing the code with rotate = 0,1, and 2. The result of changing the video rotate is that it remains rotated to landscape even after making the rotation change, the media player application also follows the rotation and when inserted into OpenFaceOffline it still cannot become portrait. Does anyone know how to solve this problem??

**Comment by kendrew06** [comment]
您的邮件我已收到，有需要会尽快回复This is an automatic reply, confirming that your e-mail was received.Thank you

**Comment by brmarkus** [comment]
Can you share the output of using e.g. `ffprobe` or `MediaInfo` for your video? What exactly is stored as metadata in your video file?
Depending on the output you might  just update the metadata without re-encode/transcode the video with 

`ffmpeg -display_rotation 90 -i input.mp4 -codec copy output.mp4`

**Comment by s160421093** [comment]
> Can you share the output of using e.g. `ffprobe` or `MediaInfo` for your video? What exactly is stored as metadata in your video file? Depending on the output you might just update the metadata without re-encode/transcode the video with
> 
> `ffmpeg -display_rotation 90 -i input.mp4 -codec copy output.mp4`

Hi, thank you for the response!

As requested, here is the MediaInfo output for the video in question (with sensitive paths and timestamps censored) from the original video (when played using a regular media player it remains portrait but when inserted into OpenFace it is rotated 90 degrees counterclockwise. 
Let me know if any additional metadata or testing output would be helpful.

Thanks again!

<details>
<summary><strong>Click here for MediaInfo Output</strong></summary>
<pre>General
Complete name               : C:\(cencored)\...\0.participant1.mov
Format                      : MPEG-4
Format profile              : QuickTime
Codec ID                    : qt   0000.00 (qt  )
File size                   : 1.06 GiB
Duration                    : 10 min 2 s
Overall bit rate mode       : Variable
Overall bit rate            : 15.2 Mb/s
Frame rate                  : 29.970 FPS
Writing library             : Apple QuickTime

Video
ID                          : 2
Format                      : AVC
Format/Info                 : Advanced Video Codec
Format profile              : High@L4
Format settings             : CABAC / 2 Ref Frames
Format settings, CABAC      : Yes
Format settings, Reference  : 2 frames
Format settings, GOP        : M=2, N=30
Codec ID                    : avc1
Codec ID/Info               : Advanced Video Coding
Duration                    : 10 min 2 s
Bit rate                    : 15.1 Mb/s
Width                       : 1 920 pixels
Height                      : 1 080 pixels
Display aspect ratio        : 16:9
Rotation                    : 90°
Frame rate mode             : Variable
Frame rate                  : 29.970 (29970/1000) FPS
Minimum frame rate          : 28.571 FPS
Maximum frame rate          : 30.000 FPS
Color space                 : YUV
Chroma subsampling          : 4:2:0
Bit depth                   : 8 bits
Scan type                   : Progressive
Bits/(Pixel*Frame)          : 0.242
Stream size                 : 1.06 GiB (99%)
Title                       : Core Media Video
Writing library             : H.264
Encoded date                : (cencored)
Tagged date                 : (cencored)
Color range                 : Limited
Color primaries             : BT.709
Transfer characteristics    : BT.709
Matrix coefficients         : BT.709
Metas                       : 3,4
Codec configuration box     : avcC

Audio
ID                          : 1
Format                      : AAC LC
Format/Info                 : Advanced Audio Codec Low Complexity
Codec ID                    : mp4a-40-2
Duration                    : 10 min 2 s
Source duration             : 10 min 2 s
Bit rate mode               : Variable
Bit rate                    : 89.2 kb/s
Channel(s)                  : 1 channel
Channel layout              : C
ChannelLayout_Original      : M
Sampling rate               : 44.1 kHz
Frame rate                  : 43.066 FPS (1024 SPF)
Compression mode            : Lossy
Stream size                 : 6.41 MiB (1%)
Source stream size          : 6.41 MiB (1%)
Title                       : Core Media Audio

Other #1
ID                          : 3
Type                        : meta
Format                      : mebx
Codec ID                    : mebx
Duration                    : 10 min 2 s
Bit rate mode               : Variable
Title                       : Core Media Metadata

Other #2
ID                          : 4
Type                        : meta
Format                      : mebx
Codec ID                    : mebx
Duration                    : 10 min 2 s
Title                       : Core Media Metadata </pre>

</details>

**Comment by brmarkus** [comment]
The metadata contains `Rotation                    : 90°`..

Have you tried calling `ffmpeg -display_rotation 90 -i input.mp4 -codec copy output.mp4` (or experiment with other values)?
Or, as mentioned earlier, `ffmpeg.exe -i input.mp4 -metadata:s:v:0 rotate=0 -y output.mp4`

**Comment by s160421093** [comment]
> The metadata contains `Rotation : 90°`..
> 
> Have you tried calling `ffmpeg -display_rotation 90 -i input.mp4 -codec copy output.mp4` (or experiment with other values)? Or, as mentioned earlier, `ffmpeg.exe -i input.mp4 -metadata:s:v:0 rotate=0 -y output.mp4`

Hello everyone,

A little update regarding the video rotation issue I previously discussed. As I mentioned, I've tried several ffmpeg commands, including:

<details>
<summary><strong>The code I've tried includes:</strong></summary>

1.  `ffmpeg -i "...\input.mov" -vf "transpose=1" -metadata:s:v:0 rotate=0 -c:a copy "...\output.mov"`
2.  `ffmpeg -i "...\input.mov" -vf "transpose=2" -metadata:s:v:0 rotate=0 -c:a copy "...\output.mov"`
3.  `ffmpeg -i "...\input.mov" -vf "transpose=2,transpose=2" -metadata:s:v:0 rotate=0 -c:a copy "...\output.mov"`
4.  `ffmpeg -i "...\input.mov" -vf "rotate=PI/2:ow=ih:oh=iw" -metadata:s:v:0 rotate=0 -c:a copy "...\output.mov"`
5.  `ffmpeg -i "...\input.mov" -vf "transpose=1,vflip" -metadata:s:v:0 rotate=0 -c:a copy "...\output.mov"`

</details>

And several other codes, but the result remained that the video was flipped into landscape mode.

I want to add that I tried the code provided by **brmarkus** a few days ago, but I've only just had the chance to provide an update now. The effective code from him is:
```shell
ffmpeg.exe -i input.mp4 -metadata:s:v:0 rotate=0 -y output.mp4
```

I've tried this code on all my videos, and it was effective. I also just realized that this code was actually suggested earlier by palich12, but I hadn't gotten around to trying it at the time.

Thank you all for your help!
